tests/test_server.py

```diff
@@ -634,6 +635,52 @@ async def test_train_with_yaml(rasa_app: SanicASGITestClient, tmp_path: Path):
     assert_trained_model(response.body, tmp_path)
 
 
+@pytest.mark.parametrize(
+    "params", [{}, {"augmentation": 20, "num_threads": 2, "force_training": True}]
+)
+async def test_train_with_yaml_with_params(
+    monkeypatch: MonkeyPatch,
+    rasa_app: SanicASGITestClient,
+    tmpdir: pathlib.Path,
```

Comment:
`tmpdir` is of type `LocalPath` which is slightly different. I suggest using `tmp_path`
```suggestion
    tmp_path: pathlib.Path,
```